components/brave_ads/core/ads_client_notifier_observer.h

```diff
@@ -0,0 +1,117 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_ADS_CORE_ADS_CLIENT_NOTIFIER_OBSERVER_H_
+#define BRAVE_COMPONENTS_BRAVE_ADS_CORE_ADS_CLIENT_NOTIFIER_OBSERVER_H_
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include "base/observer_list_types.h"
+#include "base/time/time.h"
+#include "url/gurl.h"
+
+namespace brave_ads {
+
+class AdsClientNotifierObserver : public base::CheckedObserver {
+ public:
+  AdsClientNotifierObserver();
+
+  AdsClientNotifierObserver(const AdsClientNotifierObserver& other) = delete;
+  AdsClientNotifierObserver& operator=(const AdsClientNotifierObserver& other) =
+      delete;
+
+  AdsClientNotifierObserver(AdsClientNotifierObserver&& other) noexcept =
+      delete;
+  AdsClientNotifierObserver& operator=(
+      AdsClientNotifierObserver&& other) noexcept = delete;
+
+  ~AdsClientNotifierObserver() override;
+
+  // Called when the user changes the locale of their operating system. This
+  // call is not required if the operating system restarts the browser when
+  // changing the locale. |locale| should be specified in either
+  // <ISO-639-1>-<ISO-3166-1> or <ISO-639-1>_<ISO-3166-1> format.
+  virtual void OnNotifyLocaleDidChange(const std::string& locale) {}
```

Comment:
Ideally we do need `Notify` in the observers, as this way we call `NotifyLocaleDidChange` and listen in `OnLocaleDidChange`. Same for all other observers in this file, your thoughts? Thanks

Reply:
agreed to keep it for now